src/wp-includes/class-wp-duotone.php

```diff
@@ -0,0 +1,1064 @@
+		$h = $hsla['h'];
+		$s = $hsla['s'];
+		$l = $hsla['l'];
+		$a = $hsla['a'];
+
+		$s *= ( $l < 50 ? $l : 100 - $l ) / 100;
+
+		return array(
+			'h' => $h,
+			's' => $s > 0 ? ( ( 2 * $s ) / ( $l + $s ) ) * 100 : 0,
+			'v' => $l + $s,
+			'a' => $a,
+		);
+	}
+
+	/**
+	 * Direct port of colord's hslaToRgba function.
+	 *
+	 * @see https://github.com/omgovich/colord/blob/3f859e03b0ca622eb15480f611371a0f15c9427f/src/colorModels/hsl.ts#L55
+	 *
+	 * @param array $hsla The HSLA array to convert.
+	 * @return array The RGBA array.
+	 */
+	private static function colord_hsla_to_rgba( $hsla ) {
+		return self::colord_hsva_to_rgba( self::colord_hsla_to_hsva( $hsla ) );
+	}
+
+	/**
+	 * Direct port of colord's parseHslaString function.
+	 *
+	 * @see https://github.com/omgovich/colord/blob/3f859e03b0ca622eb15480f611371a0f15c9427f/src/colorModels/hslString.ts#L17
+	 *
+	 * @param string $input The HSLA string to parse.
+	 * @return array|null An array of RGBA values or null if the RGB string is invalid.
+	 */
+	private static function colord_parse_hsla_string( $input ) {
+		// Functional syntax.
+		$is_match = preg_match(
+			'/^hsla?\(\s*([+-]?\d*\.?\d+)(deg|rad|grad|turn)?\s*,\s*([+-]?\d*\.?\d+)%\s*,\s*([+-]?\d*\.?\d+)%\s*(?:,\s*([+-]?\d*\.?\d+)(%)?\s*)?\)$/i',
+			$input,
+			$match
+		);
+
+		if ( ! $is_match ) {
+			// Whitespace syntax.
+			$is_match = preg_match(
+				'/^hsla?\(\s*([+-]?\d*\.?\d+)(deg|rad|grad|turn)?\s+([+-]?\d*\.?\d+)%\s+([+-]?\d*\.?\d+)%\s*(?:\/\s*([+-]?\d*\.?\d+)(%)?\s*)?\)$/i',
+				$input,
+				$match
+			);
+		}
+
+		if ( ! $is_match ) {
+			return null;
+		}
+
+		// For some reason, preg_match doesn't include empty matches at the end
+		// of the array, so we add them manually to make things easier later.
+		for ( $i = 1; $i <= 6; $i++ ) {
+			if ( ! isset( $match[ $i ] ) ) {
+				$match[ $i ] = '';
+			}
+		}
+
+		$hsla = self::colord_clamp_hsla(
+			array(
+				'h' => self::colord_parse_hue( $match[1], $match[2] ),
+				's' => (float) $match[3],
+				'l' => (float) $match[4],
+				'a' => '' === $match[5] ? 1 : (float) $match[5] / ( $match[6] ? 100 : 1 ),
+			)
+		);
+
+		return self::colord_hsla_to_rgba( $hsla );
+	}
+
+	/**
+	 * Direct port of colord's parse function simplified for our use case. This
+	 * version only supports string parsing and only returns RGBA values.
+	 *
+	 * @see https://github.com/omgovich/colord/blob/3f859e03b0ca622eb15480f611371a0f15c9427f/src/parse.ts#L37
+	 *
+	 * @param string $input The string to parse.
+	 * @return array|null An array of RGBA values or null if the string is invalid.
+	 */
+	private static function colord_parse( $input ) {
+		$result = self::colord_parse_hex( $input );
+
+		if ( ! $result ) {
+			$result = self::colord_parse_rgba_string( $input );
+		}
+
+		if ( ! $result ) {
+			$result = self::colord_parse_hsla_string( $input );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Take the inline CSS duotone variable from a block and return the slug. Handles styles slugs like:
```

Comment:
As per WP Documentation Standards, should be "Takes", not "Take" (third-person verb)
https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#language